app/components/user_list/index.tsx

```diff
@@ -41,29 +56,51 @@ const sectionKeyExtractor = (profile: UserProfile) => {
     return profile.username[0].toUpperCase();
 };
 
-export function createProfilesSections(profiles: UserProfile[]) {
-    const sections: {[key: string]: UserProfile[]} = {};
-    const sectionKeys: string[] = [];
-    for (const profile of profiles) {
-        const sectionKey = sectionKeyExtractor(profile);
-
-        if (!sections[sectionKey]) {
-            sections[sectionKey] = [];
-            sectionKeys.push(sectionKey);
-        }
+const sectionRoleKeyExtractor = (intl: IntlShape, cAdmin: boolean) => {
+    // Group items by channel admin or channel member
+    return intl.formatMessage(cAdmin ? messages.admins : messages.members);
+};
 
-        sections[sectionKey].push(profile);
+export function createProfilesSections(intl: IntlShape, profiles: UserProfile[], members?: ChannelMember[]) {
+    if (!profiles.length) {
+        return [];
     }
 
-    sectionKeys.sort();
+    const membersDictionary = new Map();
+    const sections = new Map();
+
+    if (members?.length) {
+        // when channel members are provided, build the sections by admins and members
+        members.forEach((m) => membersDictionary.set(m.user_id, m));
+        profiles.forEach((p) => {
+            const member = membersDictionary.get(p.id);
+            const sectionKey = sectionRoleKeyExtractor(intl, member.scheme_admin!);
+            const sectionValue = sections.get(sectionKey) || [];
 
-    return sectionKeys.map((sectionKey, index) => {
-        return {
-            id: sectionKey,
-            first: index === 0,
-            data: sections[sectionKey],
-        };
-    });
+            // combine UserProfile and ChannelMember objects so can get channel member scheme_admin permission
+            const section = [...sectionValue, {...p, ...member}];
```

Comment:
This makes so every item on the list is different after recalculating the section. Have you checked if there is any performance problem with this? (using flipper)

Reply:
I tested in `Dev Tool Discussions` which has 96 members and there were no visible delays, but created a ticket for more testing with flipper.

Ticket created for this - https://mattermost.atlassian.net/browse/MM-49673